shotover-proxy/tests/cassandra_int_tests/mod.rs

```diff
@@ -74,10 +75,14 @@ fn test_source_tls_and_single_tls() {
 #[test]
 #[serial]
 fn test_cassandra_redis_cache() {
+    let recorder = DebuggingRecorder::new();
+    let snapshotter = recorder.snapshotter();
```

Comment:
And these changes?

Reply:
Yep, I'll pull all of these out at the end once we are happy with the rest of the PR as per https://github.com/shotover/shotover-proxy/pull/661#issue-1250234838
Or would I better off completely pulling out the cache changes into a dependent PR right now and just letting people refer to the other PR to see how the new parser API works in transforms?

Reply:
I've just ignored the cache stuff for this review, but it has been helpful to see the changes illustrated.

Up to you whether you pull out the cache changes now or just before merge